### PR TITLE
Add config helpers

### PR DESCRIPTION
**Pull request by deads2k**
Adds config helpers for webconsole, service serving certs, and eventually origin

Adds serviceability for image-registry, webconsole, service serving cert, and origin.

@openshift/sig-master

### CODE REVIEW

File: pkg/config/helpers/general.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			// Don't resolve absolute paths
+			if !filepath.IsAbs(*ref) {
+				*ref = filepath.Join(base, *ref)
+			}
+		}
+	}
+	return nil
+}
+
+func MakeRelative(path, base string) (string, error) {
```
Comment (mfojtik) [inline pkg/config/helpers/general.go]: @deads2k i know you will hate me, but I think if we call this "library" we should document the exported library functions ;-)

File: pkg/config/helpers/general.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	for _, ref := range refs {
+		// Don't relativize empty paths
+		if len(*ref) > 0 {
+			rel, err := MakeRelative(*ref, base)
+			if err != nil {
+				return err
+			}
+
+			// if we have a backstep, don't mess with the path
+			if strings.HasPrefix(rel, "../") {
```
Comment (mfojtik) [inline pkg/config/helpers/general.go]: will break on windows (if we care or ever plan to use this in CLI)

File: pkg/config/helpers/serving_info.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+func GetHTTPServingInfoFileReferences(config *configv1.HTTPServingInfo) []*string {
```
Comment (mfojtik) [inline pkg/config/helpers/serving_info.go]: `var GetHTTPServingInfoFileReferences = GetServingInfoFileReferences` :-)

File: pkg/config/helpers/general.go (hunk 1 above)
Comment (deads2k) [reply pkg/config/helpers/general.go]: > @deads2k i know you will hate me, but I think if we call this "library" we should document the exported library functions ;-)

sure

File: pkg/config/helpers/general.go (hunk 2 above)
Comment (deads2k) [reply pkg/config/helpers/general.go]: > will break on windows (if we care or ever plan to use this in CLI)

Hasn't come up.  I'm inclined to leave it.  I'm also of the opinion that I only like to support forward slashes and windows users can just deal with it.

File: pkg/config/helpers/serving_info.go (hunk 3 above)
Comment (deads2k) [reply pkg/config/helpers/serving_info.go]: > var GetHTTPServingInfoFileReferences = GetServingInfoFileReferences

They are different types

**Comment by deads2k** [comment]
comments addressed

File: pkg/config/helpers/general.go (hunk 2 above)
Comment (sttts) [reply pkg/config/helpers/general.go]: Can I quote "I'm also of the opinion that I only like to support forward slashes and windows users can just deal with it." ?

**Pull request merged**